### DISCUSSION TITLE
Unable to Style canvas

### DISCUSSION THREAD

**Discussion by joshua-mcfield99** [wojtekmaj/react-pdf#1444] [Q&A] [answered]
I am using react-pdf to render a pdf file located in the public folder, I am having issues with the ability to make the canvas (being rendered by the Page component) bigger, for some reason the Page component is able to change height and width but the canvas component is not able to change height and width. Is there a way to override the style used by react-pdf_canvas without using !important?

**Comment by joshua-mcfield99** [comment]
As A little preview this works but I want to avoid using !important.
 `.react-pdf__Page__canvas {
    display: inline-block;
    width: auto !important;
    height: 100% !important;
}`

**Comment by wojtekmaj** [comment, marked as answer]
Do NOT style canvas this way. Instead, explicitly pass the `width` prop to the Page component. This is the only way to ensure that all layers stay aligned and everything looks good.

If it has to be dynamic, consider listening for window resize or container resize. I have handy hooks for this occasion - useWindowWidth and useIntersectionObserver - in my package @wojtekmaj/react-hooks, but of course, you can implement it yourself, too.
